RELEASE/scripts/autoscend/auto_quest_level_4.ash

```diff
@@ -33,8 +33,7 @@ boolean L4_batCave()
 	{
 		if (item_amount($item[Enchanted Bean]) == 0 && internalQuestStatus("questL10Garbage") < 2 && !isActuallyEd())
 		{
-			autoAdv($location[The Beanbat Chamber]);
-			return true;
+			return autoAdv($location[The Beanbat Chamber]);
```

Comment:
I'm not totally sure of the implications of this. For example, what does `autoAdv` return if there's a wandering free fight? If this is trying to fix the recent beanbat bug, this might not be the best way to handle that.

Reply:
I will test if it returns true or false when hitting a free wander. But regardless it will still work now.

```
method A
autoAdv
return true
```
is used in many many places and is just bad because it assumes success. it means any time there is a blocking condition then autoscend will hit an infinite loop. Which is exactly what just happened to me in the bat chamber, I hit an infinite loop, which this fixed.

on the other hand
```
method B
return autoAdv
```
avoids this infinite loop. If autoAdv succeeds (it only fails when there is some problem mind you.) then it will return true the same as before.

If autoAdv fails, it returns false. returning false here means it continues to the next tasks in main loop. So it will basically go do a thing elsewhere. which usually means 1 adv will be spent elsewhere. Then the loop restarts and it comes back to this spot and tries the bat chamber again. and again if it fails it will go do something else.

But there is a good chance of it not failing repeatedly. For example if the failure is caused by being too low a level, or hitting a free wander (which I am not sure actually fails), or whatever. then it will get it on the next attempt.

The worst case scenario is if you have a permanent block on a place (eg, it is out of standard and we didn't notice, for example) it is trying to adventure.

With method A, this means autoscend will always hit an infinite loop of doing nothing the moment you try to run it. And that it simply cannot be used by that person until it is fixed.

With method B, this means autoscend will always try to adventure in a permanently blocked, fail, then go adventure elsewhere.
It will successfully complete the entire run. But it will take 2x the server hits.

Usually though the block is temporary and thus responding to failure by spending 1 adventure doing something else is a good thing as it is likely the block will clear up after a few adventures

Reply:
commit dropped since you already merged a fix for this bug via #328

Reply:
I tested it using sausage goblin and some testing code.
I had it adventure in 8 bit realm. encounter a zol, returned true
then i had it adventure in 8 bit realm, encounter a sausage goblin, kill it as a free fight, return true also.